src/platform/common/stm32/bus_spi_hw.c

```diff
@@ -174,10 +174,12 @@ FAST_IRQ_HANDLER static void spiTxIrqHandler(dmaChannelDescriptor_t* descriptor)
 
 uint16_t spiCalculateDivider(uint32_t freq)
 {
-#if defined(STM32F4) || defined(STM32G4) || defined(STM32F7) || defined(APM32F4)
+#if defined(STM32F4) || defined(STM32F7) || defined(APM32F4)
     uint32_t spiClk = SystemCoreClock / 2;
 #elif defined(STM32H7)
     uint32_t spiClk = 100000000;
+#elif defined(STM32G4)
+    uint32_t spiClk = SystemCoreClock;
```

Comment:
Technically, this should be `HAL_RCC_GetPCLK1Freq` or  similar driver function. It will probably work for other processors too (checked F7), but it may be slightly complicated by SPI using both APB1 and APB2 (and thus PCLK1 / PCLK2), but that can be resolved using peripheral address. This is already implemented for timer clocks.

This special-casing will work, but it is mixing a lot of assumptions. But the simplicity may be worth it.